src/views/GameView.vue

```diff
@@ -702,6 +699,11 @@ export default {
     },
   },
   mounted() {
+    if (this.isSpectating && !this.$store.state.game.id) {
```

Comment:
Are we defensive enough if there's no game set in the store? If it's null or undefined I think this would bomb. Util we have a redirect in place for games that aren't valid or don't exist this might be an issue.

Reply:
`game` is a store module, so that should generally exist right? It's possible that it will be empty state in which case the `id` will be `null`, but I don't see how this could type error

Reply:
I couldn’t remember if we defaulted to null or an empty object, so wanted to check.

Reply:
So the game store module itself is a populated object which defaults to having empty values for the specified keys. It's controlled by the `restState()` function in the `game.js module`:

```
function resetState() {
  return {
    id: null,
    chat: [],
    deck: [],
    log: [],
    name: null,
    p0Ready: false,
    p1Ready: false,
    passes: 0,
    players: [],
    scrap: [],
    turn: 0,
    twos: [],
    myPNum: null,
    topCard: null,
    secondCard: null,
    oneOff: null,
    oneOffTarget: null,
    waitingForOpponentToCounter: false,
    myTurnToCounter: false,
    isRanked: false,
    // Threes
    waitingForOpponentToPickFromScrap: false,
    pickingFromScrap: false,
    // Fours
    discarding: false,
    waitingForOpponentToDiscard: false,
    // Sevens
    playingFromDeck: false,
    waitingForOpponentToPlayFromDeck: false,
    // Last Event
    lastEventChange: null,
    lastEventOneOffRank: null,
    lastEventTargetType: null,
    // GameOver
    gameIsOver: false,
    winnerPNum: null,
    conceded: false,
    waitingForOpponentToStalemate: false,
    consideringOpponentStalemateRequest: false,
  };
}
```

so `game.id` will be `null` by default, but the reference is typesafe